File: 2.POO/Resto/Calificacion/2°version/Java/Calificacion.java
import java.util.Scanner;
public class Calificacion {
  public static void main(String[] args) {
    Scanner mex= new Scanner(System.in);
    
     
    Calificaciones C1 = new Calificaciones();
    System.out.println("Ingresa tu nombre: ");
    C1.setNombre(mex.nextLine());
    System.out.println("Ingresa el nombre de la materia: ");
    C1.setMateria(mex.nextLine());
    System.out.println("Ingresa la calificacion del cognitivo: ");
    C1.setCognitivo(mex.nextDouble());
    System.out.println("Ingresa la calificacion del formativo: ");
    C1.setFormativo(mex.nextDouble());
    System.out.println("Ingresa la calificacion del sumativo: ");
    C1.setSumativo(mex.nextDouble());
    if(C1.Promedio() >= 7  ){
      System.out.println("\n");
      System.out.println(C1.toString());
      System.out.println("Materia aprobada");
    }else{ 
      System.out.println("\n");
      System.out.println(C1.toString());
      System.out.println("Materia no aprobada");
    }

   
    System.out.println("\n \n");
    System.out.println("---------------------------------------------------------------------------");
    System.out.println("Creador: Jesus A. Mex Damian 28/02/2023.");


  }  
}

/* 
    System.out.println("\n");

    Calificaciones C2 = new Calificaciones("Kelly Uc Panti", "contabilidad financiera");

    System.out.println(C2.nombre + " su calificacion de " + C2.materia + " es: " + 
    ((C2.cognitivo * 0.5) + (C2.formativo * 0.25) + (C2.sumativo * 0.25)));

    System.out.println("\n");

    Calificaciones C3 = new Calificaciones(8,8,8);

    System.out.println(C3.nombre + " su calificacion de " + C3.materia + " es: " + 
    ((C3.cognitivo * 0.5) + (C3.formativo * 0.25) + (C3.sumativo * 0.25)));

    System.out.println("\n");

    Calificaciones C4 = new Calificaciones("Alondra Tax Tec", "calculo integral", 10, 10, 10);

    System.out.println(C4.nombre + " su calificacion de " + C4.materia + " es: " + 
    ((C4.cognitivo * 0.5) + (C4.formativo * 0.25) + (C4.sumativo * 0.25)));
*/

File: 2.POO/Resto/Calificacion/4°version/C++/Calificacion.cpp
#include <iostream>
#include <fstream>
#include <cstdlib>
using namespace std;
#include "Calificacion.h"

int main() {

    system("color 03");
    cout << "\n \n";
    fstream archivo;
    Calificacion C1;
    string palabras, a;
    double cantidades;

    cout << "Ingresa tu nombre: ";
    getline(cin,a);
    a += ".txt";
    archivo.open(a.c_str(), ios::in);

        if(archivo.fail()){
            cerr << ">>> Nose encentra la clave del cliente " << a << ", la clave del cliente sera registrada al finalizar. \n";
        }

 
    
    cout << "Nombre de la materia: ";
    getline(cin,palabras);
    C1.setMateria(palabras);
    cout << "Ingresa la calificacion del cognitivo: ";
    cin >> cantidades;
    cin.ignore();
    C1.setCognitivo(cantidades);
    if(C1.getCognitivo() >= 0 && C1.getCognitivo() <= 10){
        cout << "Ingresa la calificacion del sumativo: ";
        cin >> cantidades;
        cin.ignore();
        C1.setSumativo(cantidades);
        if(C1.getSumativo() >= 0 && C1.getSumativo() <= 10){
            cout << "Ingresa la calificacion del formativo: ";
            cin >> cantidades;
            cin.ignore();
            C1.setFormativo(cantidades);
            if(C1.getFormativo() >= 0 && C1.getFormativo() <= 10){
                archivo.open(a.c_str(), ios::app);
                archivo << C1.estado();
                archivo.close();
            } else { cout << "Calificacion invalida"; }
        } else { cout << "Calificacion invalida"; }
    } else { cout << "Calificacion invalida"; }


    cout << "\n \n";
    cout << "------------------------------------------------------------------------------------------\n";
	cout << "Creador: Jesus A. Mex Damian 28/02/2023.";
   
    return 0;
}
 
 
 
 
/*
    cout << "\n  \n"; 

    Calificacion C2("Jesus H. Mex Arrollo", "quimica");
    cout << C2.nombre << " su calificaion de " << C2.materia << " es: " << (C2.cognitivo * 0.5) + (C2.sumativo * 0.25) + (C2.formativo * 0.25);
    
    cout << "\n  \n";   

    Calificacion C3(7, 7, 7);
    cout << C3.nombre << " su calificaion de " << C3.materia << " es: " << (C3.cognitivo * 0.5) + (C3.sumativo * 0.25) + (C3.formativo * 0.25);

    cout << "\n  \n";   

    Calificacion C4("Rubi Tilan Leon", "quimica", 7, 7, 7);
    cout << C4.nombre << " su calificaion de " << C4.materia << " es: " << (C4.cognitivo * 0.5) + (C4.sumativo * 0.25) + (C4.formativo * 0.25);
*/